src/data/queryParams.ts

```diff
@@ -0,0 +1,50 @@
+import { isEmpty, pickBy } from 'lodash'
+import { QueryParamConfigMap } from 'serialize-query-params'
+import { NumberParam, StringParam, useQueryParams } from 'use-query-params'
+
+import { CalculatorData, QueryData, defaultValues } from './calculate'
+
+const queryConfig: QueryParamConfigMap = {
+  topLocation: StringParam,
+  subLocation: StringParam,
+
+  riskProfile: StringParam,
+  interaction: StringParam,
+  personCount: NumberParam,
+
+  setting: StringParam,
+  distance: StringParam,
+  duration: NumberParam,
+  theirMask: StringParam,
+  yourMask: StringParam,
+  voice: StringParam,
+}
+
+export const filterParams = (data: CalculatorData): QueryData => {
+  const filterData = { ...data }
+  let key: keyof CalculatorData
+  for (key in filterData) {
+    if (!(key in queryConfig) || data[key] === defaultValues[key]) {
+      delete filterData[key]
+    }
+  }
+  return filterData
+}
+
+// This method chooses between existing calulator data (either defaults or from a
+// saved model) or the ones specified through query parameters.
+export const useParams = (
```

Comment:
Could we use a more descriptive function name? UseQueryDataIfPresent?

Please update the comment to describe the logic as well (e.g. "Uses queryData if any valid keys exist, otherwise returns calcData).